File: typescript/base-hmi-types/src/screens/controls/HmiSystemDiagnosisSortCriterion.ts
import { HmiAlarmSortDirection } from "./HmiAlarmSortDirection.js";
import { HmiSystemDiagnosisColumnType } from "./HmiSystemDiagnosisColumnType.js";

export class HmiSystemDiagnosisSortCriterion {
  column = HmiSystemDiagnosisColumnType.Unknown;
  sourceColumn?: string;
  direction = HmiAlarmSortDirection.Unknown;
  sourceDirection?: string;
}

File: src/BaseHmiTypes/Screens/Controls/HmiSystemDiagnosisSortCriterion.cs
namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiSystemDiagnosisSortCriterion
{
    public HmiSystemDiagnosisColumnType Column { get; set; }

    public string? SourceColumn { get; set; }

    public HmiAlarmSortDirection Direction { get; set; }

    public string? SourceDirection { get; set; }
}
